range_tombstone_list.cc

```diff
@@ -373,15 +373,24 @@ range_tombstone_list::reverter::insert(range_tombstones_type::iterator it, range
     return _dst._tombstones.insert_before(it, new_rt);
 }
 
+// Reserves space for at least one element so that subsequent v.push_back()
+// does not throw if element construction does not throw.
+// Amortizes space expansion so that a series of N calls to amortized_reserve()
+// starting from an empty container takes O(N) time.
+template <typename T>
+void amortized_reserve(T& v) {
+    v.reserve(v.size() + std::max<size_t>(1, v.size()));
```

Comment:
Agree it should receive the desired capacity, and it should also examine the existing capacity.

I think as written it is still quadratic - an {amotized_reserve, push_back}+ sequence will call reserve() with arguments of 1, 2, 4, 6, 8, 10, 12, ... 2n since size() grows as n.

Reply:
Agreed.  `amortized_reserve` should examine `capacity()` and reserve only when required for the desired new capacity.

Reply:
Right. Testing already showed a huge improvement due the switch to chunked_vector, for which reserve() is not O(N), but O(last_chunk_size).